File: expectanim/src/main/java/com/github/florent37/expectanim/core/scale/ScaleAnimExpectationValues.java
package com.github.florent37.expectanim.core.scale;

import android.support.annotation.Nullable;
import android.view.View;

/**
 * Created by florentchampigny on 17/02/2017.
 */
public class ScaleAnimExpectationValues extends ScaleAnimExpectation {

    private final float scaleX;
    private final float scaleY;

    public ScaleAnimExpectationValues(float scaleX, float scaleY, @Nullable Integer gravityHorizontal, @Nullable Integer gravityVertical) {
        super(gravityHorizontal, gravityVertical);
        this.scaleX = scaleX;
        this.scaleY = scaleY;
    }

    @Override
    public Float getCalculatedValueScaleX(View viewToMove) {
        return scaleX;
    }

    @Override
    public Float getCalculatedValueScaleY(View viewToMove) {
        return scaleY;
    }
}

File: kotlinpleaseanimate/src/main/java/com/github/florent37/kotlin/pleaseanimate/core/scale/ScaleAnimExpectationValues.kt
package com.github.florent37.kotlin.pleaseanimate.core.scale

import android.view.View

/**
 * Created by florentchampigny on 17/02/2017.
 */
class ScaleAnimExpectationValues(private val scaleX: Float, private val scaleY: Float, gravityHorizontal: Int?, gravityVertical: Int?) : ScaleAnimExpectation(gravityHorizontal, gravityVertical) {

    override fun getCalculatedValueScaleX(viewToMove: View): Float? {
        return scaleX
    }

    override fun getCalculatedValueScaleY(viewToMove: View): Float? {
        return scaleY
    }
}
